control-plane-operator/controllers/hostedcontrolplane/kas/deployment.go

```diff
@@ -140,6 +142,12 @@ func ReconcileKubeAPIServerDeployment(deployment *appsv1.Deployment,
 	}
 	auditConfigHash := util.ComputeHash(auditConfigBytes)
 
+	authConfigBytes, ok := authConfig.Data[AuthConfigMapKey]
+	if !ok {
+		return fmt.Errorf("kube apiserver authentication configuration is not expected to be empty")
```

Comment:
we allow empty/missing when this isn't configured, right?

Reply:
ah yes, this is true

Reply:
actually, this is fine.  we always create the configmap and always mount it into the KAS pod.  we only conditionally set it in the kas config.